[2.8] VMware: Add support for storagePod

Akasurde opened the pull request:
##### SUMMARY
* Added find_resource_pool_by_cluster API

Fixes: #61645

(cherry picked from commit de66abe52182bc124ffbd0ef0559375d471c79fe)
Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

##### ISSUE TYPE
<!--- Pick one below and delete the rest -->
- Bugfix Pull Request

##### COMPONENT NAME
lib/ansible/module_utils/vmware.py
lib/ansible/modules/cloud/vmware/vmware_deploy_ovf.py

Akasurde committed:
[2.8] VMware: Add support for storagePod

* Added find_resource_pool_by_cluster API

Fixes: #61645

(cherry picked from commit de66abe52182bc124ffbd0ef0559375d471c79fe)
Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

abadger commented:
This looks kind of like a new feature?

If it is okay to be backported, it needs a changelog entry.

Akasurde commented:
> This looks kind of like a new feature?

Thanks you so much for noticing this. :) 

I removed the feature which was added in 2.9 and just kept the code which is required for bug fix.
Also, added changelog fragment for bug fix, with update commit message.

Again, Thanks for help.
> If it is okay to be backported, it needs a changelog entry.

abadger commented:
Thanks @akasurde!  Merged for the 2.8.6 release
